tests/integration/csp/assertions/check.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# exit upon any error
+set -o errexit
+
+# exit upon using undeclared variables
+set -o nounset
+
+die(){
+    echo >&2 "$@"
+    exit 1
+}
+
+MODULE_NAME=$(pwd | cut -d'/' -f8)
+
+# get all the assertions (paths to texts files that we want to find in the report)
+# 1 assertion per target, one or some target per modules
+# assertions can be found in the same directory than this file
+declare -a assertions=()
+mapfile -t assertions < <(find . -name "*.txt")
+if [[ ${#assertions[@]} -eq 0 ]]; then 
+    die "Error: No assertions found in module $MODULE_NAME"
+fi
+
+# Since the assertions and the target reports must have the same name 
+# (except their extention name, .out and .txt) we extract the targets names
+# from the assertion array
+# We also use their names to log some informations so keeping an array is
+# somewhat useful
```

Comment:
```suggestion
# Since the assertions and the target reports must have the same name 
# (except their extention name, .txt and .out ) we extract the target names
# from the assertion array
# We also use their names to log some informations so keeping an array is
# somewhat useful
```